drivers/spi/spi_xlnx_axi_quadspi.c

```diff
@@ -0,0 +1,493 @@
+		xlnx_quadspi_cs_control(dev, false);
+
+		spicr = xlnx_quadspi_read32(dev, SPICR_OFFSET);
+		if ((ctx->config->operation & SPI_HOLD_ON_CS) == 0U) {
+			/* Tri-state SPI IOs */
+			spicr &= ~(SPICR_SPE);
+		}
+		xlnx_quadspi_write32(dev, spicr | SPICR_TX_FIFO_RESET,
+				     SPICR_OFFSET);
+
+		spi_context_complete(ctx, -ENOTSUP);
+	}
+
+	if (!IS_ENABLED(CONFIG_SPI_SLAVE) || !spi_context_is_slave(ctx)) {
+		/* Uninhibit master transaction */
+		spicr &= ~(SPICR_MASTER_XFER_INH);
+		xlnx_quadspi_write32(dev, spicr, SPICR_OFFSET);
+	}
+}
+
+static int xlnx_quadspi_transceive(struct device *dev,
+				   const struct spi_config *spi_cfg,
+				   const struct spi_buf_set *tx_bufs,
+				   const struct spi_buf_set *rx_bufs,
+				   bool async, struct k_poll_signal *signal)
+{
+	const struct xlnx_quadspi_config *config = dev->config;
+	struct xlnx_quadspi_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	int ret;
+
+	spi_context_lock(ctx, async, signal);
+
+	ret = xlnx_quadspi_configure(dev, spi_cfg);
+	if (ret) {
+		goto out;
+	}
+
+	spi_context_buffers_setup(ctx, tx_bufs, rx_bufs,
+				  config->num_xfer_bytes);
+
+	xlnx_quadspi_cs_control(dev, true);
+
+	xlnx_quadspi_start_tx(dev);
+
+	ret = spi_context_wait_for_completion(ctx);
+out:
+	spi_context_release(ctx, ret);
+
+	return ret;
+}
+
+static int xlnx_quadspi_transceive_blocking(struct device *dev,
+					    const struct spi_config *spi_cfg,
+					    const struct spi_buf_set *tx_bufs,
+					    const struct spi_buf_set *rx_bufs)
+{
+	return xlnx_quadspi_transceive(dev, spi_cfg, tx_bufs, rx_bufs, false,
+				       NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int xlnx_quadspi_transceive_async(struct device *dev,
+					 const struct spi_config *spi_cfg,
+					 const struct spi_buf_set *tx_bufs,
+					 const struct spi_buf_set *rx_bufs,
+					 struct k_poll_signal *signal)
+{
+	return xlnx_quadspi_transceive(dev, spi_cfg, tx_bufs, rx_bufs, true,
+				       signal);
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int xlnx_quadspi_release(struct device *dev,
+				const struct spi_config *spi_cfg)
+{
+	const struct xlnx_quadspi_config *config = dev->config;
+	struct xlnx_quadspi_data *data = dev->data;
+	uint32_t spicr;
+
+	/* Force slave select de-assert */
+	xlnx_quadspi_write32(dev, BIT_MASK(config->num_ss_bits), SPISSR_OFFSET);
+
+	/* Tri-state SPI IOs */
+	spicr = xlnx_quadspi_read32(dev, SPICR_OFFSET);
+	spicr &= ~(SPICR_SPE);
+	xlnx_quadspi_write32(dev, spicr, SPICR_OFFSET);
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static void xlnx_quadspi_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct xlnx_quadspi_config *config = dev->config;
+	struct xlnx_quadspi_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	uint32_t temp;
+	uint32_t drr;
+
+	/* Acknowledge interrupt */
+	temp = xlnx_quadspi_read32(dev, IPISR_OFFSET);
+	xlnx_quadspi_write32(dev, temp, IPISR_OFFSET);
+
+	if (temp & IPIXR_DTR_EMPTY) {
```

Comment:
I would do the inverse test here, print the message found in line 432 and return.

Would be more readable